### ISSUE TITLE
Update expo 44 stand alone app freeze splash screen

### ISSUE THREAD

**Issue by ballza95** [expo/expo#16523]
### Summary

Update expo 44 and build eas stand alone app freeze splash screen both ios and android.

But use eas build dev client working no problem but yellow log show with dev client.

RCTBridge required dispatch_sync to load RNGestureHandlerModule. This may lead to deadlocks

Please help thank you.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android, iOS

### SDK Version (managed workflow only)

44

### Environment

eas build --profile staging

### Reproducible demo

Im use monorepo

"dependencies": {
    "@config-plugins/react-native-ble-plx": "^0.0.0",
    "@date-io/date-fns": "^2.11.0",
    "@emotion/react": "11.5.0",
    "@emotion/server": "11.4.0",
    "@emotion/styled": "11.3.0",
    "@expo-google-fonts/prompt": "^0.1.0",
    "@expo/metro-config": "^0.3.11",
    "@expo/vector-icons": "^12.0.0",
    "@mui/icons-material": "^5.3.1",
    "@mui/material": "^5.3.1",
    "@react-native-async-storage/async-storage": "~1.15.0",
    "@react-native-community/datetimepicker": "4.0.0",
    "@react-native-community/masked-view": "0.1.10",
    "@react-native-firebase/app": "^12.8.0",
    "@react-native-firebase/auth": "^12.8.0",
    "@react-native-firebase/firestore": "^12.8.0",
    "@react-native-firebase/storage": "^12.8.0",
    "@react-native-picker/picker": "2.2.1",
    "@react-navigation/bottom-tabs": "^5.11.8",
    "@react-navigation/native": "^5.9.3",
    "@react-navigation/stack": "^5.14.3",
    "@reduxjs/toolkit": "^1.5.0",
    "@types/lodash": "^4.14.178",
    "@unimodules/core": "~7.2.0",
    "axios": "^0.21.1",
    "buffer": "^6.0.3",
    "convert-string": "^0.1.0",
    "core-js": "^3.6.5",
    "crypto-js": "^4.1.1",
    "date-fns": "^2.19.0",
    "expo": "^44.0.0",
    "expo-app-loading": "~1.3.0",
    "expo-asset": "~8.4.6",
    "expo-av": "~10.2.0",
    "expo-barcode-scanner": "~11.2.0",
    "expo-blur": "~11.0.0",
    "expo-camera": "~12.1.2",
    "expo-checkbox": "~2.0.0",
    "expo-constants": "~13.0.1",
    "expo-dev-client": "~0.8.4",
    "expo-device": "~4.1.0",
    "expo-file-system": "~13.1.4",
    "expo-font": "~10.0.4",
    "expo-image-manipulator": "~10.2.0",
    "expo-image-picker": "~12.0.1",
    "expo-localization": "~12.0.0",
    "expo-location": "~14.0.1",
    "expo-media-library": "~14.0.0",
    "expo-modules-core": "~0.6.4",
    "expo-notifications": "~0.14.0",
    "expo-permissions": "~13.1.0",
    "expo-sharing": "~10.1.0",
    "expo-splash-screen": "~0.14.1",
    "expo-status-bar": "~1.2.0",
    "expo-structured-headers": "2.1.0",
    "expo-updates": "~0.11.6",
    "express": "4.17.2",
    "firebase": "8.2.3",
    "firebase-admin": "^10.0.2",
    "firebase-functions": "^3.17.1",
    "geolib": "^3.3.1",
    "husky": "^6.0.0",
    "i18next": "^20.2.1",
    "list-content-ballza95": "^1.0.0",
    "lodash": "^4.17.21",
    "morgan": "^1.10.0",
    "next": "12.0.7",
    "npm": "^7.17.0",
    "prop-types": "^15.7.2",
    "react": "17.0.1",
    "react-dom": "17.0.1",
    "react-i18next": "^11.8.12",
    "react-native": "0.64.3",
    "react-native-android-open-settings": "^1.3.0",
    "react-native-animatable": "^1.3.3",
    "react-native-base64": "^0.2.1",
    "react-native-ble-manager-foxbith": "^7.6.4",
    "react-native-ble-plx": "^2.0.3",
    "react-native-button-toggle-group": "^1.1.1",
    "react-native-calendars": "^1.1254.0",
    "react-native-collapsible": "^1.6.0",
    "react-native-country-picker-modal": "^2.0.0",
    "react-native-date-picker": "^3.2.10",
    "react-native-elements": "^3.3.2",
    "react-native-geocoding": "^0.5.0",
    "react-native-gesture-handler": "~2.1.0",
    "react-native-google-places-autocomplete": "^2.2.0",
    "react-native-htmlview": "^0.16.0",
    "react-native-image-zoom-viewer": "^3.0.1",
    "react-native-iphone-x-helper": "^1.3.1",
    "react-native-keyboard-aware-scroll-view": "^0.9.3",
    "react-native-maps": "0.29.4",
    "react-native-maps-directions": "^1.8.0",
    "react-native-modal": "^11.7.0",
    "react-native-modal-datetime-picker": "^9.2.1",
    "react-native-pager-view": "5.4.9",
    "react-native-paper": "^4.7.2",
    "react-native-qrcode-svg": "^6.1.1",
    "react-native-ratings": "^8.1.0",
    "react-native-reanimated": "~2.3.1",
    "react-native-render-html": "^6.0.4",
    "react-native-safe-area-context": "3.3.2",
    "react-native-screens": "~3.10.1",
    "react-native-signature-canvas": "^3.6.0",
    "react-native-svg": "12.1.1",
    "react-native-svg-transformer": "1.0.0",
    "react-native-tab-view": "^3.0.0",
    "react-native-timeline-flatlist": "^0.7.6",
    "react-native-toast-message": "^1.4.9",
    "react-native-user-avatar": "^1.0.7",
    "react-native-vector-icons": "^8.1.0",
    "react-native-view-shot": "3.1.2",
    "react-native-web": "0.17.1",
    "react-native-webview": "11.15.0",
    "react-native-wifi-reborn": "^4.3.8",
    "react-native-youtube-iframe": "^2.0.1",
    "react-navigation": "^4.4.4",
    "react-redux": "^7.2.2",
    "react-router-dom": "^6.2.1",
    "regenerator-runtime": "0.13.7",
    "rn-swipe-button": "^1.3.4",
    "sentry-expo": "^4.0.0",
    "tslib": "^2.0.0",
    "validator": "^13.5.2",
    "victory-native": "^35.3.3"
  },

**Comment by ballza95** [comment]
I am also having this issue.

**Comment by bigxalx** [comment]
Having the same issue, but only when installing certain packages and exclusively on iOS Device. See https://github.com/expo/expo/issues/17029

**Comment by wbroek** [comment]
Got the same, out of nothing the splash screen is shown in my stand alone iOS app on the simulator (device not at this point) and I located it back to an error in my own code interacting with another native library (with expo-plugin). This interaction raised an error and I got the splash screen. Think the expo client crashed and is showing the splash screen (nothing else to show).
In the expo logs I could not see the error but when I build the app with Xcode and looked in the debugger the error from the other native library was shown. 
When solved it did not happen again

**Comment by ballza95** [comment]
> 

Thank you for solution.

How do you suggest build the app with Xcode with eas ?

**Comment by rianfloo** [comment]
Hey all,

I am facing the same issue. I am using react-native-ble-plx package. App load well in development mode on my devise and I can use it but when building it with expo build:ios and then publish it on TestFlight it is frozen on splash screen.

I made tests and I when I remove from app.json : 

```
[
        "@config-plugins/react-native-ble-plx",
        {
          "isBackgroundEnabled": true,
          "modes": ["peripheral", "central"],
          "bluetoothAlwaysPermission": "Allow $(PRODUCT_NAME) to connect to bluetooth devices",
          "bluetoothPeripheralPermission": "Allow $(PRODUCT_NAME) to connect to bluetooth devices"
        }
]
```

and comment screens importing react-native-ble-plx this works again.

Any solution?

Thanks a lot :)

**Comment by mrmuke** [comment]
Hey rianfloo, how did you fix this?

**Comment by muathime** [comment]
I had the same problem so I did code review on my code on github and found out that I was importing `useState` wrongly. Seems to have slipped on an auto import event.

`import { useState } from 'react/cjs/react.development';`
So I corrected to
`import { useState } from "react";` 

In your case it might be different but I propose searching for any import from `....react.development` and correct that.

I resolved it this way and now it works great!!!
